### ISSUE TITLE
Critical issue with image caching

### ISSUE THREAD

**Issue by khromov** [ideag/content-cards#51]
After updating to 0.9.1 and trying to add a content card, it went into an infinite loop and started downloading images infinitely. It created 1000 images before I managed to manually stop the web server.

Aside from that, these images were not properly hidden from the media library.

Here's an example:
![screen shot 2015-09-08 at 10 24 21](https://cloud.githubusercontent.com/assets/1207507/9730117/96420292-5614-11e5-951e-c9ae1a559a80.png)

I suggest retracting the 0.9.1 upgrade for now. @ideag

**Comment by khromov** [comment]
Here's the embed link I used:

http://www.meetup.com/WordPress-Stockholm/?foobar

**Comment by ideag** [comment]
@khromov sorry, but I cannot reproduce this on three of my test sites (localhost, digital ocean droplet and a local Lithuanian VPS).

**Comment by ideag** [comment]
P.S. I see you have not the latest WordPress version. Maybe that's the problem?

**Comment by khromov** [comment]
I can try to replicate with the same plugins later. The site is running 4.2.4. 

I noticed the extensions appear to be cut off. (".jpe")

Anyway, are there any loops that could potentially trigger this behaviour? Maybe we should have some sanity checking to avoid getting into infinite loops?

**Comment by ideag** [comment]
@khromov I don't believe there is a single loop in the code that runs caching. Except maybe in WordPress native functions, but those should not be a problem, too.

**Comment by sweethoneycode** [comment]
I had a client come to me with a similar problem. I think what is going on is the old images are not deleting on Cron tasks. It is hosted on SiteGround if that helps.
